packages/repository/README.md

```diff
@@ -21,47 +22,137 @@ $ npm install --save @loopback/repository
-    properties: {
-      id: {name: 'id', type: 'number', id: true},
-      title: 'string',
-      content: 'string'
-    }
-  })
-};
+  @property({id: true})
+  id: string;
+  @property() title: string;
+  @property() content: string;
+}
+```
+
+**NOTE**: There is no declarative support for data source and model yet in
+LoopBack 4. These constructs need to be created programmatically as
+illustrated above.
+
+### Defining a repository
+
+A repository can be created by extending `DefaultCrudRepository` and using
+dependency injection to resolve the datasource.
+
+```ts
+// src/repositories/note.repository.ts
+import {DefaultCrudRepository, DataSourceType} from '@loopback/repository';
+import {Note} from '../models';
+import {inject} from '@loopback/core';
+
+export class NoteRepository extends DefaultCrudRepository<
+  Note,
+  typeof Note.prototype.id
+> {
+  constructor(@inject('datasources.db') protected dataSource: DataSourceType) {
+    super(Note, dataSource);
+  }
+}
 ```
-A repository can be created directly using `DefaultCrudRepository`.
+
+### Defining a controller
+
+Controllers serve as handlers for API requests. We declare controllers as
+classes with optional dependency injection by decorating constructor parameters
+or properties.
 
 ```ts
-import {DefaultCrudRepository} from '@loopback/repository';
-// also import Note and ds from wherever you defined them
+// src/controllers/note.controller.ts
+import {repository} from '@loopback/repository';
+import {NoteRepository} from '../repositories';
+import {Note} from '../models';
+import {post, requestBody, get, param} from '@loopback/openapi-v3';
+
+export class NoteController {
+  constructor(
+    // Use constructor dependency injection to set up the repository
+    @repository(NoteRepository.name) public noteRepo: NoteRepository,
+  ) {}
+
+  // Create a new note
+  @post('/note')
+  create(@requestBody() data: Note) {
+    return this.noteRepo.create(data);
+  }
+
+  // Find notes by title
+  @get('/note/{title}')
+  findByTitle(@param.path.string('title') title: string) {
+    return this.noteRepo.find({where: {title}});
+  }
+}
+```
+
+### Run the controller and repository together
+
+#### Using the Repository Mixin for Application
+
+A Repository Mixin is available for Application that provides convenience
+methods for binding and instantiating a repository class. Bound instances can be
+used anywhere in your application using Dependency Injection.
+The `.repository(RepositoryClass)` function can be used to bind a repository
+class to an Application. The mixin will also instantiate any repositories
+declared by a component in its constructor using the `repositories` key.
+
+Repositories will be bound to the key `repositories.RepositoryClass` where
+`RepositoryClass` is the name of the Repository class being bound.
 
-  const repo = new DefaultCrudRepository(Note, ds);
+We'll use `BootMixin` on top of `RepositoryMixin` so that Repository bindings
+can be taken care of at boot.
```

Comment:
nitpick: `taken care of by @loopback/boot at boot time before the application starts` or something along those lines. Feel free to ignore.